### DISCUSSION TITLE
The derive macro `DeriveRelation` does not generate `Related` trait for entity. Is this normal ?

### DISCUSSION THREAD

**Discussion by leo91000** [SeaQL/sea-orm#752] [Q&A] [answered]
I was following the [doc to create a one to many relationship](https://www.sea-ql.org/SeaORM/docs/relation/one-to-many). When I use manual trait implementation, it works fine. However, when I'm using the macro way with `DeriveRelation`, it doesn't generate the `Related` (only the `RelationTrait` is generated) trait, is this normal or I'm having a bug ?

**Comment by billy1624** [comment, marked as answer]
Hey @leo91000, welcome and thanks for asking. `DeriveRelation` does not generate `Related` implementation for you. Take this for an example, it only implements `RelationTrait` for you.

The `DeriveRelation` macro is a simple wrapper to impl the `RelationTrait`.

```rust
#[derive(DeriveRelation)]
pub enum Relation {
    #[sea_orm(has_many = "super::fruit::Entity")]
    Fruit,
}
```

expands into

```rust
impl RelationTrait for Relation {
    fn def(&self) -> RelationDef {
        match self {
            Self::Fruit => Entity::has_many(super::fruit::Entity).into(),
        }
    }
}
```

https://www.sea-ql.org/SeaORM/docs/generate-entity/entity-structure#relation

**Reply by LasseRosenow** [reply]
Would it be technically possible to extend it to also generate the `Related` stuff? That would be sooo handy :O

**Comment by tyt2y3** [comment]
Perhaps we can have a proc macro and a DSL, although I don't like DSLs too much

```rust
related!(cake::Entity -> cake_filling::Relation::Cake.def().rev() -> cake_filling::Relation::Filling.def() -> fruit::Entity);
```

in place of

```rust
impl Related<super::filling::Entity> for Entity {
    fn to() -> RelationDef {
        super::cake_filling::Relation::Filling.def()
    }

    fn via() -> Option<RelationDef> {
        Some(super::cake_filling::Relation::Cake.def().rev())
    }
}
```

Does that actually make it more readable / understandable / easier to write?
